### PR TITLE
Add test framework: Mocha and Chai

### PR DESCRIPTION
**Pull request by koosoku**

### CODE REVIEW

File: src/models/users_model.js [hunk 1]
```diff
@@ -8,9 +8,9 @@ const connect = function(next) {
 };
 
 const selectAllUsers = function(next) {
-    const query = "SELECT id, name FROM development.users";
+    const query = "SELECT * FROM development.users";
```
Comment (singhprincejeet) [inline src/models/users_model.js]: won't that give the passwords of users in return? 
Is that a thing we should do?
Why do we have this function?

Comment (koosoku) [reply src/models/users_model.js]: We don't have a schema just yet, so its just a place holder :)

File: package.json [hunk 2]
```diff
@@ -13,10 +13,13 @@
     "async": "^2.6.0",
     "bcrypt": "^1.0.3",
     "cassandra-driver": "^3.3.0",
+    "chai": "^4.1.2",
+    "chai-http": "^3.0.0",
     "express": "^4.16.2",
```
Comment (silverAndroid) [inline package.json]: why is chai a dependency when mocha is a dev dependency

Comment (BrandonDanis) [reply package.json]: Yeah should be test

Comment (koosoku) [reply package.json]: will move it to dev dependencies

File: index.js [hunk 3]
```diff
@@ -10,5 +10,6 @@ app.get("/ping", (req, res) => {
     });
 });
 
-app.use("/users", require("./routes/users_routes"));
+app.use("/users", require("./src/routes/users_routes"));
+module.exports = app;
```
Comment (silverAndroid) [inline index.js]: Can this be on the last line? I'd prefer if it was.

File: test/controllers/users_controller_test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+const usersRoutes = require("../../index");
+const chai = require("chai"),
+    chaiHttp = require("chai-http");
+const should = chai.should();
+
+chai.use(chaiHttp);
+
+describe("Endpoints exists", () => {
```
Comment (BrandonDanis) [inline test/controllers/users_controller_test.js]: Best tests ever, making sure endpoints exist 😆

File: test/controllers/users_controller_test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                    password: "1234"
+                }).
+                end((err, res) => {
+                    should.not.exist(err);
+                    res.should.have.status(200);
+                    done();
+                });
+        });
+    });
+});
```
Comment (silverAndroid) [inline test/controllers/users_controller_test.js]: eventually these tests should check if the response body is correct

File: index.js (hunk 3 above)
Comment (koosoku) [reply index.js]: sure 👍

File: test/controllers/users_controller_test.js (hunk 4 above)
Comment (koosoku) [reply test/controllers/users_controller_test.js]: :D

File: test/controllers/users_controller_test.js (hunk 5 above)
Comment (koosoku) [reply test/controllers/users_controller_test.js]: Yup! this is just setting up the framework with examples of how to do the test. Next step is to generate properly formatted responses and testing them, as well as setting up a test db (maybe keyspace will do)

**Pull request merged**